File: languages/csharp/concept-exercises/strings/StringsTest.cs
using Xunit;

public class LogLineTest
{
    [Theory]
    [InlineData("[ERROR]: Stack overflow", "Stack overflow")]
    [InlineData("[ERROR]: Disk full", "Disk full")]
    [InlineData("[ERROR]: Segmentation fault", "Segmentation fault")]
    [InlineData("[ERROR]:  File not found ", "File not found")]
    [InlineData("[WARNING]: Disk almost full", "Disk almost full")]
    [InlineData("[WARNING]: \tTimezone not set\r\n", "Timezone not set")]
    [InlineData("[WARNING]: Unsafe password", "Unsafe password")]
    [InlineData("[INFO]: File moved", "File moved")]
    [InlineData("[INFO]: Timezone changed ", "Timezone changed")]
    public void Message(string logLine, string expected) =>
        Assert.Equal(expected, LogLine.Message(logLine));

    [Theory]
    [InlineData("[ERROR]: Stack overflow", "error")]
    [InlineData("[ERROR]: Disk full", "error")]
    [InlineData("[ERROR]: Segmentation fault", "error")]
    [InlineData("[ERROR]:  File not found ", "error")]
    [InlineData("[WARNING]: Disk almost full", "warning")]
    [InlineData("[WARNING]: \tTimezone not set\r\n", "warning")]
    [InlineData("[WARNING]: Unsafe password", "warning")]
    [InlineData("[INFO]: File moved", "info")]
    [InlineData("[INFO]: Timezone changed ", "info")]
    public void LogLevel(string logLine, string expected) =>
        Assert.Equal(expected, LogLine.LogLevel(logLine));

    [Theory]
    [InlineData("[ERROR]: Stack overflow", "Stack overflow (error)")]
    [InlineData("[ERROR]: Disk full", "Disk full (error)")]
    [InlineData("[ERROR]: Segmentation fault", "Segmentation fault (error)")]
    [InlineData("[ERROR]:  File not found ", "File not found (error)")]
    [InlineData("[WARNING]: Disk almost full", "Disk almost full (warning)")]
    [InlineData("[WARNING]: \tTimezone not set\r\n", "Timezone not set (warning")]
    [InlineData("[WARNING]: Unsafe password", "Unsafe password (warning)")]
    [InlineData("[INFO]: File moved", "File moved (info)")]
    [InlineData("[INFO]: Timezone changed ", "Timezone changed (info)")]
    public void Reformat(string logLine, string expected) =>
        Assert.Equal(expected, LogLine.Reformat(logLine));
}

File: languages/clojure/exercises/concept/strings/test/strings_test.clj
(ns strings-test
  (:require [clojure.test :refer [deftest testing is]]
            strings))

(deftest error-message-test
  (is (= "Stack overflow" (strings/message "[ERROR]: Stack overflow"))))

(deftest warning-message-test
  (is (= (strings/message "[WARNING]: Disk almost full") "Disk almost full")))

(deftest info-message-test
  (is (= (strings/message "[INFO]: File moved") "File moved")))

(deftest message-trim-test
  (is (= "Timezone not set" (strings/message "[WARNING]:   \tTimezone not set  \r\n"))))

(deftest error-log-level-test
  (is (= "error" (strings/log-level "[ERROR]: Disk full"))))

(deftest warning-log-level-test
  (is (= "warning" (strings/log-level "[WARNING]: Unsafe password"))))

(deftest info-log-level-test
  (is (= "info" (strings/log-level "[INFO]: Timezone changed"))))

(deftest error-reformat-test
  (is (= "Segmentation fault (error)" (strings/reformat "[ERROR]: Segmentation fault"))))

(deftest warning-reformat-test
  (is (= "Decreased performance (warning)" (strings/reformat "[WARNING]: Decreased performance"))))

(deftest info-reformat-test
  (is (= "Disk defragmented (info)" (strings/reformat "[INFO]: Disk defragmented"))))

(deftest reformat-trim-test
  (is (= "Corrupt disk (error)" (strings/reformat "[ERROR]: \t Corrupt disk\t \t \r\n"))))
